radicle-httpd/src/api/v1/projects.rs

```diff
@@ -422,6 +425,77 @@ async fn issue_handler(
     Ok::<_, Error>(Json(issue))
 }
 
+/// Get project patches list.
+/// `GET /projects/:project/patches`
+async fn patches_handler(
+    State(ctx): State<Context>,
+    Path(project): Path<Id>,
+    Query(qs): Query<PaginationQuery>,
+) -> impl IntoResponse {
+    let PaginationQuery { page, per_page } = qs;
+    let page = page.unwrap_or(0);
+    let per_page = per_page.unwrap_or(10);
+    let storage = &ctx.profile.storage;
+    let repo = storage.repository(project)?;
+    let patches = Patches::open(ctx.profile.public_key, &repo)?;
+    let patches = patches
+        .all()?
+        .into_iter()
+        .filter_map(|r| r.ok())
+        .map(|(id, patch, _)| {
+            json!({
+                "id": id.to_string(),
+                "author": patch.author(),
+                "title": patch.title(),
+                "description": patch.description(),
+                "state": patch.state(),
+                "target": patch.target(),
+                "tags": patch.tags().collect::<Vec<_>>(),
+                "revisions": patch.revisions().map(|(id, rev)| {
+                    json!({
+                        "id": id,
+                        "description": rev.description(),
+                    })
```

Comment:
We should move this to the json module

Reply:
yeah, my thoughts too. I will once the PR is ready.

Reply:
extracted [here](https://github.com/radicle-dev/heartwood/compare/6324139868f81933d57cf5be9d4689bf193496b3..8ffb2a155d033d75dccc3a0e708043df7a9655da#diff-c99690711746c76aea217441371c5d7b1657d4463916a2a62de0617149f93e6bR65-R84), if all else is good, I can extract json for `issue` as well

Reply:
Extracted [`api::json::issue`](https://github.com/radicle-dev/heartwood/pull/265/commits/c1ba45fcd83aecdecbdb05a60e412dc1f448ae1c) as well.